pupil-spa/src/app/login-pending/login-pending.component.ts

```diff
@@ -0,0 +1,115 @@
+import { UserService } from '../services/user/user.service';
+import { Router, ActivatedRoute } from '@angular/router';
+import { QuestionService } from '../services/question/question.service';
+import { WarmupQuestionService } from '../services/question/warmup-question.service';
+import { RegisterInputService } from '../services/register-input/registerInput.service';
+import { PupilPrefsService } from '../services/pupil-prefs/pupil-prefs.service';
+
+@Component({
+  selector: 'app-login-pending',
+  templateUrl: './login-pending.component.html',
+  styleUrls: ['./login-pending.component.scss']
+})
+export class LoginPendingComponent implements OnInit {
+
+  private submitted: boolean;
+  public loginModel = new Login('', '');
+  public loginSucceeded: boolean;
+  public connectionFailed: boolean;
+  private schoolPin: string;
+  private pupilPin: string;
+  loginPendingViewMinDisplay;
+
+  constructor(
+    private loginErrorService: LoginErrorService,
+    private loginErrorDiagnosticsService: LoginErrorDiagnosticsService,
+    private userService: UserService,
+    private router: Router,
+    private route: ActivatedRoute,
+    private questionService: QuestionService,
+    private warmupQuestionService: WarmupQuestionService,
+    private registerInputService: RegisterInputService,
+    private pupilPrefsService: PupilPrefsService
+  ) {
+    const { loginPendingViewMinDisplay } = APP_CONFIG;
+    this.loginPendingViewMinDisplay = loginPendingViewMinDisplay;
+  }
+
+  ngOnInit() {
+    const startTime = Date.now();
+    const queryParams = this.route.snapshot.queryParams;
+    this.schoolPin = queryParams && queryParams.schoolPin;
+    this.pupilPin = queryParams && queryParams.pupilPin;
+    if (this.submitted === true) {
+      return;
+    }
+    this.submitted = true;
+    this.userService.login(this.schoolPin, this.pupilPin)
+      .then(
+        async () => {
+          this.loginSucceeded = true;
+          this.connectionFailed = false;
+          this.questionService.initialise();
+          this.warmupQuestionService.initialise();
+          this.registerInputService.initialise();
+
+          const config = this.questionService.getConfig();
+          this.pupilPrefsService.loadPupilPrefs();
+          await this.displayMinTime(startTime);
+          if (config.fontSize) {
+            this.router.navigate(['font-choice']);
+          } else if (config.colourContrast) {
+            this.router.navigate(['colour-choice']);
+          } else {
+            this.router.navigate(['sign-in-success']);
+          }
+        },
+        async (err) => {
+          await this.displayMinTime(startTime);
+          this.submitted = false;
+          this.loginErrorService.changeMessage(err.message);
+          if (err.status === 401) {
+            this.loginSucceeded = false;
+            this.router.navigate(['sign-in'], { queryParams: { loginSucceeded: this.loginSucceeded } });
```

Comment:
Currently communication between login and login-pending is achieved through query params.
Wondering if we would want to change this.

Reply:
it's hard to understand why this is necessary.  surely you can manage the whole thing in a component?  

once login is clicked...
- the timer kicks in
- waits for minimum time
- checks for response
- repeats if incomplete until timeout

Reply:
Initially I thought it was a good idea to keep a separate component for consistency with the other 'pending' views in the spa app however having it under the same component surely is more simplified.
I have made the necessary changes for this